services/wallet/reader.go

```diff
@@ -0,0 +1,238 @@
+package wallet
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/services/wallet/chain"
+	"github.com/status-im/status-go/services/wallet/transfer"
+)
+
+func NewReader(s *Service, db *accounts.Database) *Reader {
+	return &Reader{s, db}
+}
+
+type Reader struct {
+	s  *Service
+	db *accounts.Database
+}
+
+type ReaderToken struct {
+	Token         *Token       `json:"token"`
+	OraclePrice   float64      `json:"oraclePrice"`
+	CryptoBalance *hexutil.Big `json:"cryptoBalance"`
+	FiatBalance   *big.Float   `json:"fiatBalance"`
+}
+
+type ReaderAccount struct {
+	Account      *accounts.Account               `json:"account"`
+	Collections  map[uint64][]*OpenseaCollection `json:"collections"`
+	Tokens       map[uint64][]*ReaderToken       `json:"tokens"`
+	Transactions map[uint64][]*transfer.View     `json:"transactions"`
+}
+
+type Wallet struct {
+	Accounts            []*ReaderAccount                 `json:"Accounts"`
```

Comment:
```suggestion
	Accounts            []*ReaderAccount                 `json:"accounts"`
```

Ignore this suggestion if this is already used by mobile